mlir/cmake/modules/MLIRDetectPythonEnv.cmake

```diff
@@ -20,13 +20,21 @@ macro(mlir_configure_python_dev_packages)
 
   find_package(Python3 ${LLVM_MINIMUM_PYTHON_VERSION}
     COMPONENTS Interpreter ${_python_development_component} REQUIRED)
+
+  # It's a little silly to detect Python a second time, but nanobind's cmake
+  # code looks for Python_ not Python3_.
+  find_package(Python ${LLVM_MINIMUM_PYTHON_VERSION}
```

Comment:
Agreed with the comment. I've run into this too but have also just done the pragmatic thing. Might be worth filing an issue in nanobind for guidance.